FirebaseDatabase/Tests/Integration/FIRDatabaseQueryTests.m

```diff
@@ -460,6 +487,39 @@ - (void)testSetLimitsWithStartAt {
   [expectations validate];
 }
 
+- (void)testSetLimitsWithStartAfter {
+  FIRDatabaseReference* ref = [FTestHelpers getRandomNode];
+  FTestExpectations* expectations = [[FTestExpectations alloc] initFrom:self];
+
+  [expectations addQuery:[[[ref queryOrderedByPriority] queryStartingAfterValue:nil]
+                             queryLimitedToFirst:1]
+         withExpectation:@{}];
```

Comment:
FWIW I find this strange. The documentation states that `nil` sorts first. Should this query match all elements (and return the first element - similar to the first case in the test above)?

Reply:
I think this is confusing because we are ordering based on priority here, which is the same for all of the nodes this test writes.

The reason that the `queryStartAtTest` above returns `a`, is that when the childKey is left `nil`, it defaults to `[Futilities minName]` here: 

https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseDatabase/Sources/Core/FQueryParams.m#L94

That's different than the intended behavior of `startAfter`. For the pagination use-case in an example like this, I'd expect the first query to use `startAt(nil)`, and subsequent queries to use `startAfter(lastReturnedPriority)`.

It might be worth removing this line from the tests to prevent this confusion in the future.

Reply:
If this is the intended behavior, then we should keep this test to ensure that we don't regress here in the future. If you are worried about the confusion that this might create, then it is better to add a comment.